src/libs/orders/components/order-shipment-card.tsx

```
@@ -5,66 +5,141 @@ import { format } from 'date-fns';
 import { FaExternalLinkAlt, FaTruck } from 'react-icons/fa';
 
 import { OrderShipment } from '@/libs/orders/types';
+import { escapeHtml } from '@/libs/common/utils';
 
-const statusColors: Record<NonNullable<OrderShipment['status']>, string> = {
+const SHIPMENT_STATUS_STYLES: Record<string, string> = {
   created: 'bg-gray-200 text-gray-800',
   in_transit: 'bg-blue-100 text-blue-800',
   delivered: 'bg-green-100 text-green-800',
   failed: 'bg-red-100 text-red-800',
 };
 
-export function OrderShipmentCard({ shipment }: { shipment: OrderShipment }) {
+export function OrderShipmentCard({ shipment, collapsed = false }: { shipment: OrderShipment, collapsed?: boolean }) {
   const {
     trackingNumber,
     courierService,
     status = 'created',
     deliveryEstimate,
     shippedAt,
     carrierTrackingUrl,
-  } = shipment;
+  } = shipment || {};
```

whyyyy??

---

dude! `shipment` is typed as required! TypeScript already guarantees that shipment cannot be `undefined` or `null`. So 

doing `shipment || {}` is unnecessary and misleading.
If `shipment` is ever `null` or `undefined`, it means something is very wrong upstream!!! either a prop validation error, a typing mistake, or a **logic bug**. Hiding that mistake with `shipment || {}` doesn't fix it; it silently fails and gives you empty values instead, which can lead to subtle UI bugs that are hard to trace.

One of the biggest advantages of TypeScript is that it forces you to handle potential null/undefined values where necessary. Circumventing that with runtime fallbacks defeats the purpose of using TypeScript in the first place.

---

I see what you mean now...
Thanks for your clarification and time. I revert it back to `shipment` instead of `shipment || {}`